src/models/hooks/useQuery/useQuery.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import { ApolloError, useApolloClient } from "@apollo/client";
+import { ErrorHandlers, handleError } from "$models/handleError";
+import { QueryOptions } from "@apollo/client/core/watchQueryOptions";
+import { defaultApolloOptions } from "./defaultApolloOptions";
+
+interface IQueryDataOptions<TData, TVariables> extends QueryOptions<TVariables, TData> {
+  errorHandlers?: ErrorHandlers;
+}
+
+/**
+ * This hook is used when no fetchMore o refetch is required.
+ * This is due to an apollo bug that involves memory leaks as explained in the following links:
+ * https://github.com/apollographql/apollo-client/pull/6216/files
+ * https://github.com/apollographql/react-apollo/issues/3402
```

Comment:
aca no linkearía sino que diría algo tipo

"Use this hook unless you need to use fetchMode, use refetch, or access the error object outside of an errorHandler. In that case, consider useAdvancedQuery."